lib/node_modules/@stdlib/lapack/base/dlange/lib/base.js

```diff
@@ -0,0 +1,367 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var dlassq = require( '@stdlib/lapack/base/dlassq' ).ndarray;
+var isRowMajor = require( '@stdlib/ndarray/base/assert/is-row-major' );
+var dasum = require( '@stdlib/blas/base/dasum' ).ndarray;
+var Float64Array = require( '@stdlib/array/float64' );
+var min = require( '@stdlib/math/base/special/min' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var abs = require( '@stdlib/math/base/special/abs' );
+
+
+// VARIABLES //
+
+var workspace = new Float64Array( 2 );
+
+
+// FUNCTIONS //
+
+/**
+* Returns the value of the one norm of a real matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @param {Float64Array} work - work array, should have `N` indexed elements if row-major layout is used
+* @param {integer} strideWork - stride length of `work`
+* @param {NonNegativeInteger} offsetWork - starting index of `work`
+* @returns {number} result
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+* var work = new Float64Array( 4 );
+*
+* var out = oneNorm( 3, 4, A, 4, 1, 0, work, 1, 0 );
+* // returns 33.0
+*/
+function oneNorm( M, N, A, strideA1, strideA2, offsetA, work, strideWork, offsetWork ) { // eslint-disable-line max-len
+	var value;
+	var temp;
+	var ia1;
+	var ia2;
+	var sum;
+	var iw;
+	var i;
+	var j;
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		iw = offsetWork;
+		for ( i = 0; i < N; i++ ) {
+			work[ iw ] = 0.0;
+			iw += strideWork;
+		}
+
+		ia1 = offsetA;
+		for ( j = 0; j < M; j++ ) {
+			ia2 = 0;
+			iw = offsetWork;
+			for ( i = 0; i < N; i++ ) {
+				work[ iw ] += abs( A[ ia1 + ia2 ] );
+				iw += strideWork;
+				ia2 += strideA2;
+			}
+			ia1 += strideA1;
+		}
+
+		value = 0.0;
+
+		iw = offsetWork;
+		for ( i = 0; i < N; i++ ) {
+			temp = work[ iw ];
+			if ( value < temp || isnan( temp ) ) {
+				value = temp;
+			}
+			iw += strideWork;
+		}
+	} else {
+		value = 0.0;
+		ia1 = offsetA;
+		for ( j = 0; j < N; j++ ) {
+			sum = dasum( M, A, strideA1, ia1 );
+			if ( value < sum || isnan( sum ) ) {
```

Comment:
If `sum` is `NaN`, we should immediately return, as `value will never be less `NaN`.